### ISSUE TITLE
About step t =1

### ISSUE THREAD

**Issue by linshuheng6** [jx-zhong-for-academic-purpose/GCN-Anomaly-Detection#5]
Hi  Jia-Xing Zhong, I have a question in this paper about step t=1.

In step t = 1, the GCN is not used, so the test result is only generated by the classifier. In the paper, the classifier is per-trained. My question is how this per-trained classifier generate good anomaly scores(in step 1, the performance is almost 77.5 on ucf-crime through TSN-RGB) with only video snippets and video-level label. In other words , I dont understand the specific process in step 1 very clearly. (Maybe I miss some key points in this paper somewhere?)
T.T
I will appreciate it if you can reply.

**Comment by jx-zhong-for-academic-purpose** [comment]
We directly use the video-level label as the supervision signals for each snippet.  To be specific, you can refer to https://github.com/yjxiong/temporal-segment-networks/blob/master/data/ucf101_splits/trainlist01.txt to understand the input format. In fact, we make no modification to the implementation of TSN and C3D. Therefore, we just briefly introduce the first step, and the detailed implementation is exactly the same as their original implementations.

**Comment by linshuheng6** [comment]
> We directly use the video-level label as the supervision signals for each snippet. To be specific, you can refer to https://github.com/yjxiong/temporal-segment-networks/blob/master/data/ucf101_splits/trainlist01.txt to understand the input format. In fact, we make no modification to the implementation of TSN and C3D at the first step. Therefore, we just briefly introduce the first step, and the detailed implementation is exactly the same as their original implementations.

I'm very glad to receive your reply!
You mean the model in step 1 is only trained by snippet and its corresponding video-level label then this model is used to test on UCF-crime.  Do I understand that correctly?
By the way, I also have several doubts about the experimental details.
1. Is the number (3) of snippets in a video as same as it in the original TSN? 
2. Is the number of snippets the same in these two models (C3D and TSN)? 
3. The ground truth of test video is given as the form of frame-level. How do you transform it to snippet-level ground truth? 
Thank you for your time!
Best wished!

**Comment by jx-zhong-for-academic-purpose** [comment]
You understand correctly.
As for the details:
1. Not exactly the same as the original TSN, and we mainly utilize hyper-parameters from its TPAMI-2018 version, https://arxiv.org/abs/1705.02953. In my mind, it should be 7 or 9. 
2. Not the same, the input unit of TSN is 5?10? frames and that of C3D is 16 frames. For short videos (eg. UCSD-Peds with only  about100 frames), it matters more,  and using the same number is not a good choice.
3. We simply duplicate the snippet-level ground truth into frame-level scores, as the authors of UCF-Crime do https://github.com/WaqasSultani/AnomalyDetectionCVPR2018/blob/master/Evaluate_Anomaly_Detector.m.

**Comment by linshuheng6** [comment]
> You understand correctly.
> As for the details:
> 
> 1. Not exactly the same as the original TSN, and we mainly utilize hyper-parameters from its TPAMI-2018 version, https://arxiv.org/abs/1705.02953. In my mind, it should be 7 or 9.
> 2. Not the same, the input unit of TSN is 5?10? frames and that of C3D is 16 frames. For short videos (eg. UCSD-Peds with only  about100 frames), it matters more,  and using the same number is not a good choice.
> 3. We simply duplicate the snippet-level ground truth into frame-level scores, as the authors of UCF-Crime do https://github.com/WaqasSultani/AnomalyDetectionCVPR2018/blob/master/Evaluate_Anomaly_Detector.m.

Thank you very much !!

**Comment by poweryin** [comment]
@jx-zhong-for-academic-purpose .Hi,I have two questiones. 
1.When step =1, input snippet and its corresponding video-level label ，in other words，input the normal snippets（in normal video） and the corresponding label=0, and the snippets of abnormal video and the corresponding label=1. However, for the abnormal video, the video level label 1 is not be used（only used the normal snippets of normal video and its corresponding label 0），because this will affect the parameter update of the classifier.I don't know if I understand it right.Looking forward to your reply.
2.Is this pre-trained classifier a feature extraction module or an anomaly detection module, and what data is used for pre-training?
Looking forward to your reply.
Thanks
Best wishes

**Comment by poweryin** [comment]
@jx-zhong-for-academic-purpose .Hi,I have two questiones.1.When step =1, input snippet and its corresponding video-level label ，in other words，input the normal snippets（in normal video） and the corresponding label=0, and the snippets of abnormal video and the corresponding label=1. However, for the snippets of the abnormal video, the video level label 1 is not be used（only used the normal snippets of normal video and its corresponding label 0），because this will affect the parameter update of the classifier.I don't know if I understand it right.Looking forward to your reply.
2.Is this pre-trained classifier a feature extraction module or an anomaly detection module, and what data is used for pre-training?
Looking forward to your reply.
Thanks
Best wishes
--

**Comment by poweryin** [comment]
@jx-zhong-for-academic-purpose .Hi,I have two questiones.
1.  When step =1, input snippet and its corresponding video-level label ，in other words，input the normal snippets（in normal video） and the corresponding label=0, and the snippets of abnormal video and the corresponding label=1. However, for the snippets of the abnormal video, the video level label 1 is not be used（only used the normal snippets of normal video and its corresponding label 0），because this will affect the parameter update of the classifier.I don't know if I understand it right.Looking forward to your reply.
2.  Is this pre-trained classifier a feature extraction module or an anomaly detection module, and what data is used for pre-training?
Looking forward to your reply.
Thanks
Best wishes

**Comment by jx-zhong-for-academic-purpose** [comment]
1. All labels are taken into consideration, which may introduce predictive noises. How to clean the noises is one of the key contributions of this paper.
2. We have elaborated the experiment section of our paper.

**Comment by poweryin** [comment]
Thanks for your patience very much.I'm very glad to receive your reply in time!
1. That is, when t=1, the snippets and the corresponding video-level labels are  input into the classifier, and a rough probability estimate is obtained. When t>=2, the video-level label given in the first step is no longer used. ,Is that so？
2. Sorry, I ignored this detail.  I thought that after pre-training the feature extraction module, that the classifier was also pre-trained with UCF-crime dataset and video-level label. Now I feel that there is no need for pre-training the UCF-crime before t=1.
Looking forward to your reply.
Thanks
Best wishes

**Comment by jx-zhong-for-academic-purpose** [comment]
Good Luck~

**Comment by jx-zhong-for-academic-purpose** [comment]
1. Yes， as shown in Fig1.
2. The pre-training can boost the performance as many researchers point out.

**Comment by poweryin** [comment]
Thank you for your patience. It solved my confusion, thank you very much.
Best wishes
